### PR TITLE
Create separate directories for v1 and v2 plutus scripts

### PR DESCRIPTION
**Pull request by Jimbo4350**
Refactor plutus-example to make it easier to add future versions of plutus
Add an example plutus v2

### CODE REVIEW

**Comment by james-iohk** [comment]
What do we do about existing Alonzo scripts that we want to compile as V2? I made this comment on #482: It would be good to be able to compile all existing V1 scripts as V2 without so much code repetition. Perhaps it would be better to keep just V2 functionality in the V2 dir and add V2 serialisation to the existing scripts? To make it less confusing, directories could be named after era and export multiple versions where supported.

File: plutus-example/src/PlutusExample/PlutusVersion1/RedeemerContextScripts.hs [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+plutusV1RedeemerContextTestScript = Plutus.unValidatorScript validator
+
+pv1RedeemerContextTestScriptBs :: SBS.ShortByteString
+pv1RedeemerContextTestScriptBs = SBS.toShort . LB.toStrict $ serialise plutusV1RedeemerContextTestScript
+
+scriptContextTextPayingScript :: PlutusScript PlutusScriptV1
+scriptContextTextPayingScript = PlutusScriptSerialised pv1RedeemerContextTestScriptBs
+
+
+-- Minting script that checks the minting value, validty interval and
```
Comment (newhoggy) [inline plutus-example/src/PlutusExample/PlutusVersion1/RedeemerContextScripts.hs:177]: `s/validty/validity`

**Review by newhoggy — APPROVED** [review]

File: plutus-example/test/Test/PlutusExample/Gen.hs [hunk 2]
```diff
@@ -16,8 +16,9 @@ import Cardano.Ledger.Era qualified as Ledger
 import Cardano.Ledger.Shelley.UTxO qualified as Ledger
 import Cardano.Ledger.TxIn qualified as Ledger
 import Gen.Cardano.Api.Typed
-import Plutus.V1.Ledger.Api qualified as Plutus hiding (singleton)
-import Plutus.V1.Ledger.Interval qualified as Plutus
+import Ledger qualified as Plutus
```
Comment (koslambrou) [inline plutus-example/test/Test/PlutusExample/Gen.hs:19]: You don't seem to need this.

File: plutus-example/plutus-example.cabal [hunk 3]
```diff
@@ -77,6 +82,7 @@ library
                       , containers
                       , ouroboros-consensus
                       , ouroboros-network
+                      , plutus-ledger
```
Comment (koslambrou) [inline plutus-example/plutus-example.cabal:85]: You shouldn't need this if I'm not mistaken.

**Review by koslambrou — APPROVED** [review]

**Pull request merged**